frontend/src/components/urlInput/urlInput.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useRef, useState } from "react";
+import { Button, Form, InputGroup } from "react-bootstrap";
+import { Control, useFieldArray } from "react-hook-form";
+
+import { useSites, ValidSiteTypeEnum } from "src/graphql";
+
+const CLASSNAME = "URLInput";
+
+interface URLInputProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  control: Control<any>;
+  type: ValidSiteTypeEnum;
+}
+
+const URLInput: React.FC<URLInputProps> = ({ control, type }) => {
+  const {
+    fields: urls,
+    append,
+    remove,
+  } = useFieldArray<
+    { urls: Array<{ url: string; site: { id: string; name: string } }> },
+    "urls",
+    "key"
+  >({
+    control,
+    name: "urls",
+    keyName: "key",
+  });
+  const [newURL, setNewURL] = useState("");
+  const selectRef = useRef<HTMLSelectElement | null>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const { data, loading } = useSites();
+
+  if (loading) return <></>;
+  const sites = (data?.querySites.sites ?? []).filter((s) =>
+    s.valid_types.includes(type)
+  );
+
+  const cleanURL = (
+    regexStr: string | undefined | null,
+    url: string
+  ): string | undefined => {
+    if (!regexStr) return;
+
+    const regex = new RegExp(regexStr);
+    const match = regex.exec(url);
+
+    return match?.[1];
+  };
+
+  const handleAdd = () => {
+    if (!inputRef.current) return;
+    const selectedSiteID = selectRef.current?.value;
+    const selectedSite = sites.find((s) => s.id === selectedSiteID);
+    const cleanedURL = cleanURL(selectedSite?.regex, inputRef.current.value);
+
+    if (selectedSite)
+      append({
+        url: cleanedURL ?? inputRef.current.value,
+        site: selectedSite,
+      });
```

Comment:
In case it wasn't intended: after successfully adding a link, the url input and site select are not reset.